### PR TITLE
wl-clicker: init at v0.3.1

### PR DESCRIPTION
**Pull request by Flameopathic**
## Description of changes
Packaging of autoclicker program for Wayland

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6703,6 +6703,12 @@
     name = "Sebastian Neubauer";
     keys = [ { fingerprint = "2F93 661D AC17 EA98 A104  F780 ECC7 55EE 583C 1672"; } ];
   };
+  Flameopathic = {
+    email = "flameopathic@gmail.com";
+    github = "Flameopathic";
+    githubID = 64027365;
```
Comment (nevivurn) [inline maintainers/maintainer-list.nix:6709]:
```suggestion
    githubId = 64027365;
```

Seems to be a typo, this is what is causing the CI failures.

**Review by nevivurn — COMMENTED** [review]
Also, the commits need to be squashed.

**Comment by Flameopathic** [comment]
@nevivurn all should be fixed now, thank you for the comments!

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Wayland autoclicker";
+    longDescription = "Script for auto clicking at incredibly high speeds - user must be a part of `input` group to run.";
+    homepage = "https://github.com/phonetic112/wl-clicker";
+    license = licenses.mit;
+    maintainers = [ maintainers.Flameopathic ];
+    mainProgram = "wl-clicker";
+    platforms = platforms.all;
```
Comment (nevivurn) [inline pkgs/by-name/wl/wl-clicker/package.nix:47]: It currently fails on aarch64-linux. We should either fix fix the build or mark as broken. Also, I assume 

```suggestion
    platforms = platforms.unix;
```

is more appropriate here.

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  buildInputs = [
+    wlr-protocols
+    wayland
+  ];
+
+  src = fetchFromGitHub {
+    owner = "phonetic112";
+    repo = "wl-clicker";
+    rev = "v${finalAttrs.version}";
+    sha256 = "0ddi383ilxqm0nvn05asn6m77sgi96fw07np95wx089a97b0mr8y";
```
Comment (nevivurn) [inline pkgs/by-name/wl/wl-clicker/package.nix:24]: use `hash = "sha256-..."`

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install build/wl-clicker $out/bin/wl-clicker
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (nevivurn) [inline pkgs/by-name/wl/wl-clicker/package.nix:40]:
```suggestion
  meta = {
```

(And adjust below lines as relevant)
See #292468

**Review by nevivurn — CHANGES_REQUESTED** [review]

File: pkgs/by-name/wl/wl-clicker/package.nix (hunk 2 above)
Comment (Flameopathic) [reply pkgs/by-name/wl/wl-clicker/package.nix:47]: would `platforms = lib.intersectLists (lib.platforms.linux) (lib.platforms.x86_64);` work for solving these issues, for now? I don't think i'm at the stage where i know what's going wrong so i can fix it.

Comment (nevivurn) [reply pkgs/by-name/wl/wl-clicker/package.nix:47]: If the package is supposed to work on said platforms, but is broken, we should use `broken = ...;`. If it is not supposed to work at all, then set it in platforms.

If it doesn't build on darwin (I haven't checked) and aarch64-linux, you can use `broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;`

Comment (Flameopathic) [reply pkgs/by-name/wl/wl-clicker/package.nix]: okay, thank you.

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  meta = {
+    description = "Wayland autoclicker";
+    longDescription = "Script for auto clicking at incredibly high speeds - user must be a part of `input` group to run.";
+    homepage = "https://github.com/phonetic112/wl-clicker";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.Flameopathic ];
+    mainProgram = "wl-clicker";
+    platforms = lib.platforms.linux;
+    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
```
Comment (nevivurn) [inline pkgs/by-name/wl/wl-clicker/package.nix:48]: If excluding darwin from platforms was intended, then you can omit darwin here. Either specify platforms.unix in platforms or simplify the `broken =` line.

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  buildInputs = [
+    wlr-protocols
+    wayland
+  ];
+
+  src = fetchFromGitHub {
+    owner = "phonetic112";
+    repo = "wl-clicker";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-0ddi383ilxqm0nvn05asn6m77sgi96fw07np95wx089a97b0mr8y";
```
Comment (nevivurn) [inline pkgs/by-name/wl/wl-clicker/package.nix:24]: Sorry, I didn't quite explain how to get the proper SRI hash.
You cannot use the base32 hash as-is, you can convert between them like so:

```console
$ nix-hash --to-sri --type sha256 0ddi383ilxqm0nvn05asn6m77sgi96fw07np95wx089a97b0mr8y
sha256-HuUK1kkqIdB5SdcewJ1J8elzqrFaFWC3BRV3GgcasTU=
```

File: pkgs/by-name/wl/wl-clicker/package.nix (hunk 5 above)
Comment (Flameopathic) [reply pkgs/by-name/wl/wl-clicker/package.nix]: it was; understood. thank you for taking me through this process in such detail, sorry for the nubile errors. adjusted.

File: pkgs/by-name/wl/wl-clicker/package.nix (hunk 6 above)
Comment (Flameopathic) [reply pkgs/by-name/wl/wl-clicker/package.nix]: that's okay, i should've read up on it further. adjusted.

**Review by nevivurn — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -6703,6 +6703,12 @@
     name = "Sebastian Neubauer";
     keys = [ { fingerprint = "2F93 661D AC17 EA98 A104  F780 ECC7 55EE 583C 1672"; } ];
   };
+  Flameopathic = {
```
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:6706]: This should be a separate commit.

Comment (Flameopathic) [reply maintainers/maintainer-list.nix]: handled in separate PR, addition no longer necessary; removed it from this PR :+1:

File: pkgs/by-name/wl/wl-clicker/package.nix [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  meta = {
+    description = "Wayland autoclicker";
+    longDescription = "Script for auto clicking at incredibly high speeds - user must be a part of `input` group to run.";
+    homepage = "https://github.com/phonetic112/wl-clicker";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.Flameopathic ];
+    mainProgram = "wl-clicker";
+    platforms = lib.platforms.linux;
+    broken = stdenv.isLinux && stdenv.isAarch64;
```
Comment (FliegendeWurst) [inline pkgs/by-name/wl/wl-clicker/package.nix:48]:
```suggestion
    broken = stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isAarch64;
```

**Review by FliegendeWurst — APPROVED** [review]
